driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -156,15 +186,21 @@ public void onResult(final List<RawBsonDocument> rawDocuments, final Throwable t
     }
 
     private void resumeableOperation(final AsyncBlock asyncBlock, final SingleResultCallback<List<RawBsonDocument>> callback) {
+        synchronized (this) {
+            isTrue("open", !isClosed.get() || closePending.get());
```

Comment:
I doubt we want to throw an `IllegalStateException` here if this assertion fails as the callback will never return.

I also don't see an equivalent check in the sync version - so perhaps this is not needed?

Reply:
I have updated this to call `onResult` with a `MongoException`.